src/jit/SimdInl.h

```diff
@@ -311,3 +311,25 @@ static void emitGlobalSet128(sljit_compiler* compiler, Instruction* instr)
     sljit_emit_op1(compiler, SLJIT_MOV, SLJIT_R0, 0, SLJIT_MEM1(SLJIT_R0), context->globalsStart + globalSet->index() * sizeof(void*));
     sljit_emit_simd_mov(compiler, SLJIT_SIMD_STORE | SLJIT_SIMD_REG_128 | SLJIT_SIMD_ELEM_128, src.arg, SLJIT_MEM1(SLJIT_R0), JITFieldAccessor::globalValueOffset());
 }
+
+static void emitSelect128(sljit_compiler* compiler, Instruction* instr)
+{
+    Operand* operands = instr->operands();
+    assert(instr->opcode() == ByteCode::SelectOpcode && instr->paramCount() == 3);
+
+    JITArg args[2];
+    JITArg target(operands + 3);
+    JITArg cond(operands + 2);
+    simdOperandToArg(compiler, operands, args[0], SLJIT_SIMD_ELEM_128, SLJIT_FR2);
+    simdOperandToArg(compiler, operands + 1, args[1], SLJIT_SIMD_ELEM_128, SLJIT_FR1);
+    sljit_s32 type = SLJIT_SIMD_REG_128;
+    if (SLJIT_IS_MEM(target.arg)) {
+        type |= SLJIT_SIMD_STORE | SLJIT_SIMD_ELEM_128;
+    }
+    sljit_jump* lbl_if_true = sljit_emit_cmp(compiler, SLJIT_EQUAL, cond.arg, cond.argw, SLJIT_IMM, 0);
```

Comment:
the varaible naming style is different here. Furthermore this is only needed when `type` is not -1, see emitFloatSelect

Reply:
What does -1 mean in this context?  I couldn't find it in the sljit docs.  Based on how it's used elsewhere (and what my mentor said) I inferred that it *might* be something like an undefined type.

Reply:
See emitFloatSelect I mentioned above:
https://github.com/Samsung/walrus/blob/main/src/jit/FloatMathInl.h#L342

The select might receive its condition in a flag register, which is represented by type. Otherwise it produces the condition (type -1). This is not related to sljit.

Reply:
For example in `emitFloatSelect` it's unclear what the significance of the type is, because if it's anything other than `-1` it won't even emit the comparison, so only `-1` makes sense as a value.  Are other types going to be handled later?

When we were reading the codebase with my mentor, we came to the conclusion that since only `-1` is ever passed as a type, we can omit it.

Reply:
Oh, okay, @kulcsaradam helped out a bit and now I think I get it.  So that tells sljit of the operation, not the operand.  I thought it was similar to

Reply:
The -1 is something which cannot be used as a comparison type:
https://github.com/zherczeg/sljit/blob/master/sljit_src/sljitLir.h#L1505

So type can have 35 different values (maybe more later), but they are all greater than 0.

Reply:
Oh, I see.  Thanks!  But would any other comparison make sense?  What should the code do if `type != -1`?

Reply:
Nothing. The `type` is simply the parameter of `sljit_emit_jump`.

Reply:
Right now that's just a plain `SLJIT_JUMP`, should the caller of `simdSelect128` be the one deciding what kind of jump to emit?

Reply:
Right, so now it emits a conditional jump if `type != -1`.